### PR TITLE
consuming multiple files model

### PR DESCRIPTION
**Pull request by nemni8**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -15,7 +15,8 @@
         "sourceType": "module"
     },
     "rules": {
-        "import/no-extraneous-dependencies": "error"
+        "import/no-extraneous-dependencies": "error",
+        "curly": "error"
```
Comment (dlvhdr) [inline .eslintrc:19]: niceee

File: scripts/ast-patches/utils.ts [hunk 2]
```diff
@@ -8,6 +8,10 @@ export const is$wModule = (
   ts.isModuleDeclaration(statement) &&
   statement.name.text === Constants.$w_MODULE_NAME;
 
+export const isGlobalModule = (
+  statement: ts.Statement
+): statement is ts.ModuleDeclaration =>
+  ts.isModuleDeclaration(statement) && statement.name.text === "global";
```
Comment (dlvhdr) [inline scripts/ast-patches/utils.ts:14]: can you explain what this global is?

File: scripts/ast-patches/utils.ts [hunk 3]
```diff
@@ -24,3 +28,11 @@ export const isInterfaceMixinMember = (
     ts.isIdentifier(member.name) &&
     member.name.escapedText === name
   );
+
+export const fetch$wModuleFromGlobal = (
+  statement: ts.ModuleDeclaration
+): ts.ModuleDeclaration | undefined => {
+  if (statement.body && ts.isModuleBlock(statement.body)) {
+    return statement.body.statements.find(is$wModule);
+  }
```
Comment (dlvhdr) [inline scripts/ast-patches/utils.ts:37]: I would return undefined explicitly

File: scripts/ast-patches/main.ts [hunk 4]
```diff
@@ -18,26 +19,56 @@ const transformer = (/*context*/) => (
...
+  const tsProgram = ts.createProgram(sourcefiles, {});
+  const sourceAst = sourcefiles.map(filePath => {
+    return {
+      path: filePath,
+      ast: tsProgram.getSourceFile(filePath)
+    };
+  });
+  if (!sourceAst) {
+    return;
+  }
```
Comment (dlvhdr) [inline scripts/ast-patches/main.ts:42]: can this maybe be a function? not critical

File: scripts/constants.ts [hunk 5]
```diff
@@ -1,26 +1,57 @@
...
   DECLARATIONS_DTS_FILENAME
 );
-const EVENT_HANDLERS_JSON_FILENAME = "eventHandlers.json";
-const $w_MODULE_NAME = "$w";
+const DECLARATIONS_DTS_ROOT_PATH_DEPRECATED = path.join(
+  DEST_TYPES_FOLDER,
+  DECLARATIONS_DTS_FILENAME_DEPRECATED
+);
 
+const $w_MODULE_NAME = "$w";
```
Comment (dlvhdr) [inline scripts/constants.ts:27]:
```suggestion
const $W_MODULE_NAME = "$w";
```

File: scripts/ast-patches/allowAnyPropOnPlayableMixinNextPrevious.ts [hunk 6]
```diff
@@ -1,67 +1,80 @@
...
+  mixinName: string;
+  memberName: string;
+}): void => {
+  const globalStatment = ast.statements.find(isGlobalModule);
+  const $w = globalStatment
+    ? fetch$wModuleFromGlobal(globalStatment)
+    : ast.statements.find(is$wModule);
+  if (!$w || !$w.body || !ts.isModuleBlock($w.body)) {
+    return;
+  }
```
Comment (dlvhdr) [inline scripts/ast-patches/allowAnyPropOnPlayableMixinNextPrevious.ts:29]: These lines repeat themselves a lot, can you extract to a function?

File: src/dynamicTypes/typesLoader.ts [hunk 7]
```diff
@@ -24,10 +24,21 @@ const dynamicallyImportFullCorvidTypes = async (): Promise<{
...
+}> => {
+  return import(
+    /* webpackChunkName: "fullCorvidTypesJSON.deprecated" */ "../../dist/fullCorvidTypes.deprecated.json"
+  );
+};
+
 export const getCollectionLibs = async (
-  key: LibCollections
+  key: LibCollections,
+  { multipleFiles }: { multipleFiles?: boolean } = {}
```
Comment (dlvhdr) [inline src/dynamicTypes/typesLoader.ts:37]: Why do it in an object? I think it's more readable if outside, but your call.
Anyway, cosider extracting into a type and giving a default value.
```suggestion
  { multipleFiles }: SomeType = {multipleFiles: false}
```

File: src/index.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ import {
 } from "./dynamicTypes";
 
 const { getCollectionLibs, LibCollections } = typesLoader;
-
+const { TS_CONFIG_PATHS, BASE_LIBS } = Constants;
```
Comment (dlvhdr) [inline src/index.ts:15]: why not at the top?

File: test/utils/tsconfig.ts [hunk 9]
```diff
@@ -0,0 +1,89 @@
+import path from "path";
```
Comment (dlvhdr) [inline test/utils/tsconfig.ts:1]: noice 🚀

**Review by dlvhdr — CHANGES_REQUESTED** [review]
minor comments, really nice job!

File: scripts/ast-patches/utils.ts (hunk 2 above)
Comment (nemni8) [reply scripts/ast-patches/utils.ts:14]: we need to wrap $w namespace with global in order for it to be available without importing it.

File: scripts/ast-patches/utils.ts [hunk 10]
```diff
@@ -1,17 +1,26 @@
 import ts from "typescript";
 import Constants from "../constants";
 export type Writable<T> = { -readonly [P in keyof T]: T[P] };
+export interface ModuleDeclarationWithModuleBlock extends ts.ModuleDeclaration {
```
Comment (dlvhdr) [inline scripts/ast-patches/utils.ts:4]: nice

**Review by dlvhdr — APPROVED** [review]

**Pull request merged**